wordpress-mobile/WordPress-Android
# My Site Menu: Plugins item - Update visibility rule

Fixes #19240 

This PR fixes the issue where the My Site "plugins" menu item was showing for users who did not have manage options capabilities.

**To test:**
- Install the app
- Login 

**User does not have manage options capabilities for selected site**
- Select a site that is [JP connected, uses the wp com rest api, and the user does not have manage options capabilities (use editor)]
- Navigate to My Site > Menu tab
- ✅ Verify the "plugins" item is not shown under the Manage header

**User has manage options**
- Select a site that is JP connected, is using the wp com rest api, and the user does have manage options capabilities (use admin)
- Navigate to My Site > Menu tab
- ✅ Verify the "plugins" item is shown under the Manage header

**Self hosted site**
- Select a site that is self hosted and you have admin capabilities
- Navigate to My Site > Menu tab
- ✅ Verify the "plugins" item is shown under the Manage header

## Regression Notes
1. Potential unintended areas of impact
The "plugins" item shows when it shouldn't or it doesn't show when it should

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:

UI Changes testing checklist: N/A
